src/SonarQube.Client/Api/Requests/V7_20/GetIssuesRequest.cs

```diff
@@ -35,26 +36,41 @@ public class GetIssuesRequest : PagedRequestBase<SonarQubeIssue>, IGetIssuesRequ
 
         protected override string Path => "api/issues/search";
 
-        protected override SonarQubeIssue[] ParseResponse(string response) =>
-            JObject.Parse(response)["issues"]
+        protected override SonarQubeIssue[] ParseResponse(string response)
+        {
+            var root = JObject.Parse(response);
+
+            // Lookup component key -> path for files. Each response contains normalized data, containing
+            // issues and components, where each issue's "component" property points to a component with
+            // the same "key". We obtain the FilePath of each issue from its corresponding component.
+            var componentKeyPathLookup = GetComponentKeyPathLookup(root);
+
+            return root["issues"]
                 .ToObject<ServerIssue[]>()
-                .Select(ToSonarQubeIssue)
+                .Select(issue => ToSonarQubeIssue(issue, componentKeyPathLookup))
                 .ToArray();
+        }
+
+        private static ILookup<string, string> GetComponentKeyPathLookup(JObject root)
+        {
+            var components = root["components"] == null
+                ? Array.Empty<ServerComponent>()
+                : root["components"].ToObject<ServerComponent[]>();
 
-        private static SonarQubeIssue ToSonarQubeIssue(ServerIssue issue) =>
-            new SonarQubeIssue(ComputePath(issue), issue.Hash, issue.Line, issue.Message, ComputeModuleKey(issue), 
+            return components
+                .Where(c => c.IsFile)
+                .ToLookup(c => c.Key, c => c.Path); // Using a Lookup because it does not throw, unlike the Dictionary
+        }
+
+        private static SonarQubeIssue ToSonarQubeIssue(ServerIssue issue, ILookup<string, string> componentKeyPathLookup) =>
+            new SonarQubeIssue(ComputePath(issue, componentKeyPathLookup), issue.Hash, issue.Line, issue.Message, ComputeModuleKey(issue),
                 GetRuleKey(issue.CompositeRuleKey), issue.Status == "RESOLVED");
 
-        private static string ComputePath(ServerIssue issue) =>
-            // Component is "{SubProject}:Path"
-            issue.SubProject != null
-                ? issue.Component.Substring(issue.SubProject.Length + 1)
-                : string.Empty;
+        private static string ComputePath(ServerIssue issue, ILookup<string, string> componentKeyPathLookup) =>
+            componentKeyPathLookup[issue.Component].FirstOrDefault() ?? string.Empty;
```

Comment:
The actual fix is here. Instead of substring-ing `issue.Component` we look up the SonarQube-provided path in the `components` collection of the response.